main/acle.md

```diff
@@ -4932,6 +5044,1381 @@ assertions, which may speed up access to aligned data (and will fault
+    to 1, otherwise the behavior is undefined.
+
+    The inline asm can indicate that it reads the current contents of ZA
+    and/or that it changes the contents of ZA by adding `"za"` to the
+    asm's clobber list. Using the clobber list for this purpose is a
+    syntactic convenience: it does not fit the normal semantics for clobbers.
+
+    If the inline asm does not have a `"za"` clobber but nevertheless
+    reads the current contents of ZA or changes the contents of ZA,
+    the behavior is undefined.
+
+2.  If F does not have [ZA state](#za-state), the inline asm must
+    “comply with the lazy saving scheme”, in the sense of
+    [[AAPCS64]](#AAPCS64). The behavior in other cases is undefined.
+
+    The inline asm is [ill-formed](#ill-formed) if it has a `"za"` clobber.
+
+## SME attributes
+
+All of the attributes described in this section can be specified
+using the GNU `__attribute__` syntax. Their names can be used
+directly or with two underscores added to each side. For example:
+
+``` c
+  __attribute__((arm_streaming))
+  __attribute__((__arm_streaming__))
+```
+
+Specifying an attribute multiple times is equivalent to specifying it once.
+
+Some of the attributes described in this section apply to function types.
+Their semantics are as follows:
+
+*   If the attribute is attached to a function declaration or definition,
+    it applies to the type of the function.
+
+*   If the attribute is attached to a function type, it applies to that type.
+
+*   If the attribute is attached to a pointer to a function type, it applies
+    to that function type.
+
+*   Otherwise, the attribute is [ill-formed](#ill-formed).
+
+Except where noted otherwise, function types that have an attribute are
+incompatible with function types that do not. For example:
+
+``` c
+  #define ATTR __attribute__((arm_streaming))
+
+  typedef void (*n_callback_type)(void);
+  n_callback_type n_callback_ptr;
+  void n_extern_function(void);
+  void n_local_function(void) { ... }
+
+  typedef ATTR void (*s_callback_type)(void);
+  s_callback_type s_callback_ptr;
+  ATTR void s_extern_function(void);
+  ATTR void s_local_function(void) { ... }
+
+  void foo() {
+    n_callback_ptr = n_callback_ptr;     // OK
+    n_callback_ptr = n_extern_function;  // OK
+    n_callback_ptr = n_local_function;   // OK
+    n_callback_ptr = s_callback_ptr;     // Ill-formed
+    n_callback_ptr = s_extern_function;  // Ill-formed
+    n_callback_ptr = s_local_function;   // Ill-formed
+
+    s_callback_ptr = n_callback_ptr;     // Ill-formed
+    s_callback_ptr = n_extern_function;  // Ill-formed
+    s_callback_ptr = n_local_function;   // Ill-formed
+    s_callback_ptr = s_callback_ptr;     // OK
+    s_callback_ptr = s_extern_function;  // OK
+    s_callback_ptr = s_local_function;   // OK
+  }
+```
+
+The function type attributes cannot be used with K&R-style
+“unprototyped” C function types. For example:
```

Comment:
Would it make sense to add a footnote to explain or point at the definition of K&R-style
“unprototyped” C function types?

Reply:
I'll add a description to the section that defines “abstract machine” etc. (with a link).

Reply:
Now done.  (Missed this in the previous commit, sorry.)